### PR TITLE
Mie scattering properties computer

### PR DESCRIPTION
**Pull request by nollety**
Resolves eradiate/eradiate-issues#109

# Description

Adds the possibility to compute the Mie scattering properties from micro-physical properties input (radius distribution, refractive index) at any wavelength.

# Checklist

- [ ] The code follows the relevant coding guidelines
- [x] The code generates no new warnings
- [ ] The code is appropriately documented
- [x] The code is tested to prove its function
- [ ] The feature branch is rebased on the current state of the `main` branch
- [x] I give permission that the Eradiate project may redistribute my contributions under the terms of its license

### CODE REVIEW

File: eradiate/radprops/mie.py [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+
+_DEFAULT_MU = np.linspace(-1, 1, 201)
+
+
+@ureg.wraps(ret=None, args=("nm", None, None, "dimensionless"), strict=False)
+def compute_properties(
+    w: t.Union[ureg.Quantity, np.ndarray],
+    rdist: xr.DataArray,
+    m: xr.DataArray,
+    mu: t.Union[ureg.Quantity, np.ndarray] = _DEFAULT_MU,
```
Comment (leroyvn) [inline eradiate/radprops/mie.py:22]: You should use `np.typing.ArrayLike` for params accepting anything that can be converted to a Numpy array. Be sure to double check in that makes sense; you'd then have:
```suggestion
    w: t.Union[ureg.Quantity, np.typing.ArrayLike],
    rdist: xr.DataArray,
    m: xr.DataArray,
    mu: t.Union[ureg.Quantity, np.typing.ArrayLike] = _DEFAULT_MU,
```

File: eradiate/radprops/mie.py [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+    m: :class:`~xarray.DataArray`
+        Complex refractive index.
+
+    mu: :class:`~pint.Quantity` or :class:`~numpy.ndarray`
+        Scattering angle cosines [dimensionless].
+
+    Returns
+    -------
+    :class:`~xarray.DataArray`
+        Radiative properties.
```
Comment (leroyvn) [inline eradiate/radprops/mie.py:45]: We have built-in aliases for many of these (see `docs/conf.py`). In that case:
```suggestion
    w : quantity or array-like
        Wavelength values [nm].

    rdist : DataArray
        Particle radius probability distribution.

    m : DataArray
        Complex refractive index.

    mu : quantity or array-like
        Scattering angle cosines [dimensionless].

    Returns
    -------
    DataArray
        Radiative properties.
```

**Review by leroyvn — COMMENTED** [review]
Hi @nollety, just a couple of quick comments about type hints and docstrings.

File: eradiate/radprops/mie.py (hunk 2 above)
Comment (nollety) [reply eradiate/radprops/mie.py:45]: Nice, thanks!

**Comment by nollety** [comment]
Marked as private feature.

**Pull request closed without merging**